presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java

```diff
@@ -115,19 +119,61 @@ public ScalarFunctionImplementation specialize(BoundVariables boundVariables, in
         List<Object> extraParameters = computeExtraParameters(matchingMethodsGroup.get(), context);
         MethodHandle matchingMethodHandle = applyExtraParameters(matchingMethod.get(), extraParameters);
 
-        return new ScalarFunctionImplementation(nullableResult, nullableArguments, matchingMethodHandle, deterministic);
+        return new ScalarFunctionImplementation(nullableResult, nullableArguments, getIsNullAnnotations(matchingMethod.get()),
+                matchingMethodHandle, deterministic);
+    }
+
+    private static boolean hasIsNullAnnotation(Annotation[] annotations)
```

Comment:
This class is introduced by Teradata to handle decimal-related functions (as you've probably realized). This framework is not annotation-based. I would expand `nullableArguments` in `SqlScalarFunctionBuilder` to store an enum (NON_NULLABLE, NULLABLE_BOXED, NULLABLE_FLAG) instead.